#! /usr/bin/env python

PACKAGE='urg_node'

from dynamic_reconfigure.parameter_generator_catkin import *

from math import pi

gen = ParameterGenerator()
#       Name                    Type      Reconfiguration level             Description                                                                                              Default Min   Max
gen.add("tf_prefix",             str_t,    0,                               "tf_prefix for this node's output.",                                                                     "")
gen.add("frame_id",              str_t,    0,                               "Output frame_id for the laserscan.",                                                                    "laser")
gen.add("time_offset",           double_t, 0,                               "A manually calibrated offset (in seconds) to add to the timestamp before publication of a message.",    0.0,   -10.0, 10.0)
gen.add("angle_min",             double_t, 1,                               "Controls the angle of the first range measurement in radians.",                                         -pi,     -pi,   pi)
gen.add("angle_max",             double_t, 1,                               "Controls the angle of the last range measurement in radians.",                                           pi,     -pi,   pi)
gen.add("range_min",             double_t, 1,                               "Controls the range of the first range measurement in radians.",                                         0.02,     0.02,   2)
gen.add("cluster",               int_t,    1,                               "The number of adjacent range measurements to cluster into a single reading.",                           1,         1,   99)
gen.add("skip",                  int_t,    1,                               "The number of input messages to skip between each output message.",                                     0,         0,    9)

exit(gen.generate(PACKAGE, "urg_node", "URG"))
